docs/components/icons/Example.vue

```diff
@@ -1,31 +1,133 @@
 <script setup>
-  import * as icons from '@warp-ds/icons/vue';
+import decamelize from "decamelize";
+import * as icons from "@warp-ds/icons/vue";
+import * as reactIcons from "@warp-ds/icons/react";
+import { wModal } from "@warp-ds/vue";
+import { ref } from "vue";
 
-  const mappedIconsBySize = Object.keys(icons).reduce((acc, current) => {
-    const matches = current.match(/\d+$/g);
-      
-    acc[matches[0]] = acc[matches[0]] ? {...acc[matches[0]], ...{[current]: icons[current]}} : {[current]: icons[current]};
+const showModal = ref(false);
+let modalData = ref({
+  iconName: "",
+  react: "",
+  reactSyntax: "",
+  vue: "",
+  vueSyntax: "",
+  elements: "",
+  elementsIcon: "",
+});
 
-    return acc;
-  }, {});
-  
-  const getIconName = (icon) => icon.replace(/Icon|\d+/g, '');
+const mappedIconsBySize = Object.keys(icons).reduce((acc, current) => {
+  const matches = current.match(/\d+$/g);
+
+  acc[matches[0]] = acc[matches[0]]
+    ? { ...acc[matches[0]], ...{ [current]: icons[current] } }
+    : { [current]: icons[current] };
+
+  return acc;
+}, {});
+
+const getIconName = (icon) => icon.replace(/Icon|\d+/g, "");
+
+const setIconData = (fullName, event) => {
+  const reactIconName = Object.keys(reactIcons)
+    .find((icon) => icon === fullName)
+    .replace("Icon", "");
+
+  const outputString = decamelize(reactIconName, { separator: "-" }).replace(
+    /([a-zA-Z])(\d+)/,
+    "$1-$2"
+  );
+
+  modalData = {
+    iconName: fullName,
+    react: `import ${fullName} from '@warp-ds/icons/react/${outputString}';`,
+    reactSyntax: `<${fullName} />`,
+    vue: `import ${fullName} from '@warp-ds/icons/vue/${outputString}';`,
+    vueSyntax: `<icon-${outputString} />`,
+    elements: `import from '@warp-ds/icons/elements/${outputString}';`,
+    elementsSyntax: `<w-icon-${outputString}></w-icon-${outputString}>`,
+  };
+};
+
+const reset = () => {
+  modalData = {
+    iconName: "",
+    react: "",
+    reactSyntax: "",
+    vue: "",
+    vueSyntax: "",
+    elements: "",
+    elementsSyntax: "",
+  };
+};
 </script>
 
 <template>
-  <section v-for="size in Object.keys(mappedIconsBySize)" :key="size" style="background-color: var(--vp-c-bg-alt);" class="rounded-8 p-24 mb-24">
+  <section
+    v-for="size in Object.keys(mappedIconsBySize)"
+    :key="size"
+    style="background-color: var(--vp-c-bg-alt)"
+    class="rounded-8 p-24 mb-24"
+  >
     <h2 style="color: var(--vp-c-text-1)">{{ size }}</h2>
+    <w-modal
```

Comment:
I also noticed that this modal seems to have a darker or double-overlay, compared to the modal's overlay under [Overlays/Modal](https://warp-ds.github.io/tech-docs/components/modal/): 
![Skärmavbild 2023-11-28 kl  14 20 12](https://github.com/warp-ds/tech-docs/assets/144792260/5828a8ac-1d08-4750-bce5-a5bf92bb019e)

Reply:
Weird, it's the same component